File: 2017-minimum-number-of-flips-to-make-the-binary-string-alternating/minimum-number-of-flips-to-make-the-binary-string-alternating.java
class Solution {
    public int minFlips(String s) {
        int a[] = new int[2];
        char[] c = s.toCharArray();
        int ans = 100000, x = 0, y = 0;
        for(int i = 0; i < c.length; i++, x ^= 1) {
            a[(c[i] - '0') ^ x]++;
        }
        for(int i = 0; i < c.length; i++, x ^= 1, y ^= 1) {
            a[(c[i] - '0') ^ y]--;
            a[(c[i] - '0') ^ x]++;
            ans = Math.min(ans, Math.min(a[0], a[1]));
        }
        return ans;
    }
}

File: 2017-minimum-number-of-flips-to-make-the-binary-string-alternating/minimum-number-of-flips-to-make-the-binary-string-alternating.js
var minFlips = function(s) {
    const a = [0, 0];
    let ans = 100000, x = 0, y = 0;
    for(let i = 0; i < s.length; i++, x ^= 1) {
        a[(s[i] - '0') ^ x]++;
    }
    for(let i = 0; i < s.length; i++, x ^= 1, y ^= 1) {
        a[(s[i] - '0') ^ y]--;
        a[(s[i] - '0') ^ x]++;
        ans = Math.min(ans, Math.min(a[0], a[1]));
    }
    return ans;
};